### ISSUE TITLE
docker image expects database name to be postgresql

### ISSUE THREAD

**Issue by hazcod** [temporalio/temporal#877]
## Expected Behavior
temporal should not hardcode database hostnames since most people will want to dedicate a custom name to the database, such as `temporal-db`.

## Actual Behavior
temporal `start.sh` script expects the database name to be `postgresql` for the right seeds & wait sequence to be selected. Better would be to check for `POSTGRES_SEEDS` being non-empty or an additional argument `DB_TYPE-postgresql`. This is worsened by the fact that link aliases in Docker are being deprecated, not allowing for service names to be overwritten in Docker.

## Steps to Reproduce the Problem
```yaml
  temporal:
    image: temporalio/auto-setup:1.0.0
    restart: "on-failure:10"
    security_opt:
    - "no-new-privileges"
    cap_drop:
    - ALL
    tty: false
    networks:
    - backend
    environment:
    - "LOG_LEVEL=warn"
    - "AUTO_SETUP=true"
    - "DB=temporaldb"
    - "DB_PORT=26257"
    - "POSTGRES_USER=root"
    - "POSTGRES_PWD=postgres"
    - "POSTGRES_SEEDS=postgresql"
    - "SKIP_DEFAULT_NAMESPACE_CREATION=true"
    ports:
    - 7233
    depends_on:
    - temporaldb
```

Logs:
```
+ cqlsh --cqlversion=3.4.4
Connection error: ('Unable to connect to any servers', {'127.0.0.1:9042': error(111, "Tried connecting to [('127.0.0.1', 9042)]. Last error: Connection refused")})
+ echo 'waiting for cassandra to start up'
waiting for cassandra to start up
+ sleep 1

```

## Specifications
https://github.com/temporalio/temporal/blob/master/docker/start.sh#L147

**Comment by wxing1292** [comment]
Hi

**PostgreSQL is not officially supported yet (will be in few days)**

There are few docker / startup script changes related to PostgreSQL, plz wait for the official release

On master branch:
```zsh
docker build . -t temporalio/auto-setup:local --build-arg TARGET=auto-setup
```
then use yaml
```yaml
version: "3"
services:
  temporaldb:
    image: postgres:9.6
    ports:
      - "5432:5432"
    environment:
      POSTGRES_USER: temporal
      POSTGRES_PASSWORD: temporal
  temporal:
    image: temporalio/auto-setup:${SERVER_TAG:-1.1.0}
    ports:
      - "7233:7233"
    volumes:
      - ${DYNAMIC_CONFIG_DIR:-../config/dynamicconfig}:/etc/temporal/config/dynamicconfig
    environment:
      - "DB=postgresql"
      - "DB_PORT=5432"
      - "POSTGRES_USER=temporal"
      - "POSTGRES_PWD=temporal"
      - "POSTGRES_SEEDS=temporaldb"
      - "DYNAMIC_CONFIG_FILE_PATH=config/dynamicconfig/development.yaml"
    depends_on:
      - temporaldb
```
along with 
```zsh
SERVER_TAG=local docker-compose -f docker/docker-compose-postgres.yml up
```
works

**Comment by ivanr** [comment]
@wxing1292 We would be interested in using Temporal with Postgres, but, if I am reading this correctly, you appear to be targeting 9.6, which is ancient? There have been at least four major Postgres releases since (10-13), with tremendous improvements. Perhaps 13 is too recent (last month), but 12 wouldn't be controversial :)

**Comment by hazcod** [comment]
@ivanr IMO postgres support has been working fine for a while, I've been using cockroachdb.
Albeit there is the limitation this ticket is about.

**Comment by mnichols** [comment]
This is probably known, but when i followed @wxing1292 steps above on fresh `master` pull this morning, cassandra is still being attempted for connection over and over. Disabling it in the startup script and rebuilding didnt seem to make it go away:
```
+ echo 'waiting for cassandra to start up'
temporal_1           | + sleep 1
temporal_1           | waiting for cassandra to start up
temporal_1           | + cqlsh --cqlversion=3.4.4
temporal_1           | Connection error: ('Unable to connect to any servers', {'127.0.0.1:9042': error(111, "Tried connecting to [('127.0.0.1', 9042)]. Last error: Connection refused")})
temporal_1           | + echo 'waiting for cassandra to start up'
temporal_1           | waiting for cassandra to start up
temporal_1           | + sleep 1
temporal_1           | + cqlsh --cqlversion=3.4.4
```

**Comment by wxing1292** [comment]
solved in 1.2.1 release
https://github.com/temporalio/temporal/releases/tag/v1.2.1
